python/src/flow_graph.cpp

```diff
@@ -584,4 +598,63 @@ add_flow_graph_bindings(py::module& m)
 
                      return repr;
                  });
+
+    pyfgraph.def_property_readonly("size", &fs::py_flow_graph::size);
+    pyfgraph.def_property_readonly("grid_shape", &fs::py_flow_graph::grid_shape);
+    pyfgraph.def(
+        "apply_kernel",
+        [](fs::py_flow_graph& flow_graph,
+           py::object flow_kernel,
+           py::object flow_kernel_data) -> int
+        {
+            flow_kernel_data.attr("check_bindings")();
+            auto kernel = flow_kernel.attr("kernel").cast<fs::PyNumbaFlowKernel>();
+            auto kernel_data
+                = flow_kernel_data.attr("jitclass_ptr").cast<fs::PyNumbaFlowKernelData>();
+
+            if (kernel.n_threads == 1)
+            {
+                auto py_apply_kernel
+                    = py::module::import("fastscapelib").attr("flow").attr("py_apply_kernel");
+                return py_apply_kernel(flow_graph, flow_kernel, flow_kernel_data).cast<int>();
+            }
+            else
+            {
+                py::gil_scoped_release release;
+                auto fs_kernel = (fs::NumbaFlowKernel&) kernel;
+                auto fs_kernel_data = (fs::NumbaFlowKernelData&) kernel_data;
+
+                return flow_graph.apply_kernel(fs_kernel, fs_kernel_data);
+            }
+        });
+
+    py::enum_<fs::kernel_application_order>(
+        m, "KernelApplicationOrder", py::arithmetic(), "Order to satisfy to apply a kernel.")
```

Comment:
```suggestion
    py::enum_<fs::flow_graph_traversal_order>(
        m, "FlowGraphTraversalOrder", py::arithmetic(), "Order in which to visit the flow graph nodes.")
```

A bit more generic in case we want to reuse it elsewhere later. (same suggestion for the other relevant places).